simulator-spring-boot/src/test/java/org/citrusframework/simulator/http/HttpOperationScenarioIT.java

```diff
@@ -0,0 +1,244 @@
+package org.citrusframework.simulator.http;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+import io.apicurio.datamodels.openapi.models.OasOperation;
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.citrusframework.context.TestContext;
+import org.citrusframework.exceptions.CitrusRuntimeException;
+import org.citrusframework.exceptions.TestCaseFailedException;
+import org.citrusframework.functions.DefaultFunctionRegistry;
+import org.citrusframework.http.actions.HttpServerResponseActionBuilder;
+import org.citrusframework.http.message.HttpMessage;
+import org.citrusframework.log.DefaultLogModifier;
+import org.citrusframework.message.Message;
+import org.citrusframework.openapi.OpenApiRepository;
+import org.citrusframework.openapi.actions.OpenApiClientResponseActionBuilder;
+import org.citrusframework.openapi.model.OasModelHelper;
+import org.citrusframework.simulator.config.SimulatorConfigurationProperties;
+import org.citrusframework.simulator.scenario.ScenarioEndpoint;
+import org.citrusframework.simulator.scenario.ScenarioEndpointConfiguration;
+import org.citrusframework.simulator.scenario.ScenarioRunner;
+import org.citrusframework.spi.Resources.ClasspathResource;
+import org.citrusframework.util.FileUtils;
+import org.citrusframework.validation.DefaultMessageHeaderValidator;
+import org.citrusframework.validation.DefaultMessageValidatorRegistry;
+import org.citrusframework.validation.context.HeaderValidationContext;
+import org.citrusframework.validation.json.JsonMessageValidationContext;
+import org.citrusframework.validation.json.JsonTextMessageValidator;
+import org.citrusframework.validation.matcher.DefaultValidationMatcherRegistry;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.http.HttpHeaders;
+import org.springframework.test.util.ReflectionTestUtils;
+
+@ExtendWith(MockitoExtension.class)
+class HttpOperationScenarioIT {
```

Comment:
this is not an integration test, or is it? or do you refer to the integration with "real" files?